timesketch/lib/analyzers/browser_search.py

```diff
@@ -0,0 +1,219 @@
+            Google Sites: https://sites.google.com/site/.*/system/app/pages/
+                          search?q=query
+            DuckDuckGo: https://duckduckgo.com/?q=query
+            Yahoo: https://search.yahoo.com/search?p=query
+            Yahoo: https://search.yahoo.com/search;?p=query
+            Yandex: https://www.yandex.com/search/?text=query
+            YouTube: https://www.youtube.com/results?search_query=query
+
+        Args:
+          url (str): URL.
+          parameter (str): the parameter that contains the search query.
+
+        Returns:
+          str: search query, the search parameter or None if no
+              query was found.
+        """
+        if '{0:s}='.format(parameter) not in url:
+            return None
+
+        return self._get_url_parameter_value(url, parameter)
+
+    def _get_url_parameter_value(self, url, parameter):
+        """Retrieves the GET parameter from a URL.
+
+        Args:
+          url (str): URL.
+          parameter (str): the parameter to extract.
+
+        Returns:
+          str: the GET parameter value or None if no parameter was found.
+        """
+        # Make sure we're analyzing the query part of the URL.
+        _, _, url = url.partition('?')
+        # Look for a key value pair named 'q'.
+        _, _, url = url.partition('{0:s}='.format(parameter))
+        if not url:
+            return ''
+
+        # Strip additional key value pairs.
+        parameter, _, _ = url.partition('&')
+        parameter = parameter.replace('+', ' ')
+
+        return self._decode_url(parameter)
+
+    def run(self):
+        """Entry point for the browser search analyzer.
+
+        Returns:
+            String with summary of the analyzer result
+        """
+        query = 'source_short:"WEBHIST"'
+        return_fields = ['message', 'url']
+
+        # Generator of events based on your query.
+        events = self.event_stream(
+            query_string=query, return_fields=return_fields)
+
+        search_emoji = '&#x1F50E'
+
+        for event in events:
+            url = event.source.get('url')
+            message = event.source.get('message')
+
+            if url is None:
+                continue
+
+            for engine, expression, method_name, parameter in self._URL_FILTERS:
+                callback_method = getattr(self, method_name, None)
+                if not callback_method:
+                    continue
+
+                match = expression.search(url)
+                if not match:
+                    continue
+
+                if parameter:
+                    search_query = callback_method(url, parameter)
+                else:
+                    search_query = callback_method(url)
+
+                if not search_query:
+                    continue
+
+                event.add_attributes(
+                    {'search_string': search_query,
+                     'human_readable': '[{0:s}] Search: {1:s} - {2:s}'.format(
```

Comment:
You can replace the human readable with:
`event.set_human_readable(<human_readable_string>)`

You don't need to have the original message in the string IMO, the attribute is still there if anyone want to see it.

Maybe something like:
`human_readable = '{0:s} search: {1:s}'.format(engine, search_query)`

Reply:
ack

Reply:
As discussed offline, OK to keep the message in the human_readable